### ISSUE TITLE
'x' is missing in props validation when extending interface and another type present with 'Prop' in it's name

### ISSUE THREAD

**Issue by aximusnl** [yannickcr/eslint-plugin-react#2853]
When Props are extending an interface that is imported AND another type is present with the name 'Prop' in it, react/proptypes is throwing a false positive. 

```
eslint-plugin-react@7.21.5
```

**How to reproduce**

This is OK
``` tsx
interface PropsBase {
	foo: string;
}

type RenderProps  = {
	something: string;
}

interface Props extends PropsBase {
	bar: string;
}

const MyComponent = ({ foo, bar}: Props): JSX.Element => {
	return (
		<div>
			{foo} {bar}
		</div>
	)
}

export {
	MyComponent,
}
```

This shows an error: 'foo' is missing in props validation

PropsBase.ts
``` tsx
interface PropsBase {
	foo: string;
}
```

MyComponent.tsx
```tsx
import { PropsBase } from './PropsBase';

interface Props extends PropsBase {
	bar: string;
}

type RenderProps  = {
	something: string;
}

const MyComponent = ({ foo, bar}: Props): JSX.Element => {
	return (
		<div>
			{foo} {bar}
		</div>
	)
}

export {
	MyComponent,
}
```

**Comment by ljharb** [comment]
This is expected - it’s the limits of static analysis. The linter can’t read other files, so it has no idea that foo exists in the other file.

**Comment by RemyMachado** [comment]
I'm getting this error without extending another type. I get it from the components created from a freshly installed storybook.js.

```javascript
import React from "react";
import "./button.css";

export interface ButtonProps {
  /**
   * Is this the principal call to action on the page?
   */
  primary?: boolean;
  /**
   * What background color to use
   */
  backgroundColor?: string;
  /**
   * How large should the button be?
   */
  size?: "small" | "medium" | "large";
  /**
   * Button contents
   */
  label: string;
  /**
   * Optional click handler
   */
  onClick?: () => void;
}

/**
 * Primary UI component for user interaction
 */
export const Button: React.FC<ButtonProps> = ({
  primary = false,
  size = "medium",
  backgroundColor,
  label,
  ...props
}) => {
  const mode = primary
    ? "storybook-button--primary"
    : "storybook-button--secondary";
  return (
    <button
      type="button"
      className={["storybook-button", `storybook-button--${size}`, mode].join(
        " "
      )}
      style={{ backgroundColor }}
      {...props}
    >
      {label}
    </button>
  );
};
```

Why was #2654 closed then ?
